dask/dataframe/methods.py

```diff
@@ -253,10 +257,7 @@ def concat(dfs, axis=0, join='outer', uniform=False, filter_warning=True):
 @concat_dispatch.register((pd.DataFrame, pd.Series, pd.Index))
 def concat_pandas(dfs, axis=0, join='outer', uniform=False, filter_warning=True):
     if axis == 1:
-        with warnings.catch_warnings():
-            if filter_warning:
-                warnings.simplefilter('ignore', FutureWarning)
-            return pd.concat(dfs, axis=axis, join=join)
+        return pd.concat(dfs, axis=axis, join=join, **concat_kwargs)
```

Comment:
> Some context: pd.concat used to sort the non-axis axis (so sort columns when concating vertically) if they weren't already aligned.

So if I do the equivalent of `np.vstack` (getting taller dataframes) pandas will sort the names of my columns?  And if I do the equivalent of `np.hstack` (getting wider dataframes) then pandas will sort my index?  

I'm surprised by the former (assuming that the dataframes agree on column order) and we probably can't do the latter efficiently.

Reply:
Only if the non-expanding axis isn't already aligned.

```python
In [4]: a = pd.DataFrame({'c': [1, 2], 'a': [3, 4]})

In [5]: b = pd.DataFrame({'c': [1, 2]})

In [6]: pd.concat([a, b])
/Users/taugspurger/.virtualenvs/pandas-dev/bin/ipython:1: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version
of pandas will change to not sort by default.

To accept the future behavior, pass 'sort=False'.

To retain the current behavior and silence the warning, pass 'sort=True'.

  #!/Users/taugspurger/Envs/pandas-dev/bin/python3
Out[6]:
     a  c
0  3.0  1
1  4.0  2
0  NaN  1
1  NaN  2
```

That sorted since `a.columns` wasn't aligned with `b.columns`. It was an old implementation detail that pandas 0.23 fixed (with a deprecation cycle for people relying on the behavior).

> And if I do the equivalent of np.hstack (getting wider dataframes) then pandas will sort my index?

Correct, though people didn't like this behavior, so I'd say we don't need to match it if we don't already. Not sorting is the future behavior we want to match.

Reply:
And keep in mind that the sorting happens if and only if the indexes aren't already aligned. We can't know that ahead of time for `axis=1`. `dd.concat([a, b], axis=1, sort=True)` seems like it's hard to do, but also not useful. We could certainly raise in that case.

Reply:
OK, I'll give this a shot

Reply:
Ok, so supporting `sort=` seems doable, but is not trivial due to all the different ways in which we can concat.  If it's ok I'm inclined to merge what's here to avoid warnings and then raise an issue about adding sort support for `concat`.  Thoughts @TomAugspurger ?